sdk/macros.cmake

```diff
@@ -95,11 +95,18 @@ macro(add_eyrie_runtime target_name tag plugins) # the files are passed via ${AR
   list(APPEND PLUGIN_FLAGS "-DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}")
   list(APPEND PLUGIN_FLAGS "-DCMAKE_OBJCOPY=${CMAKE_OBJCOPY}")
 
+  get_filename_component(SRCDIR ${CMAKE_SOURCE_DIR} NAME)
+  if(${SRCDIR} STREQUAL "sdk")
+    set(EYRIE_SRCDIR ${CMAKE_SOURCE_DIR}/../runtime)
+  elseif(${SRCDIR} STREQUAL "keystone")
+    set(EYRIE_SRCDIR ${CMAKE_SOURCE_DIR}/runtime)
+  else()
+    message(FATAL_ERROR "Don't know how to find runtime from current directory" ${SRCDIR})
+  endif()
+
   ExternalProject_Add(eyrie-${target_name}
     PREFIX ${runtime_prefix}
-    GIT_REPOSITORY https://github.com/keystone-enclave/keystone-runtime
-    GIT_TAG ${tag}
-    UPDATE_COMMAND git fetch
+    DOWNLOAD_COMMAND rm -rf ${eyrie_src} && cp -ar ${EYRIE_SRCDIR} ${eyrie_src}
```

Comment:
Quick question: what happens if someone modifies the runtime source code for a certain binary by directly modifying the files in `${eyrie_src}`, and then run make? Will it remove all the changes and re-sync the directory with ${EYRIE_SRCDIR} ?

Reply:
From my experience running this code, this source copying only happens at cmake initialization time. Therefore, any changes in `${eyrie_src}` would persist through most of the lifetime of a user's build directory. Of course, the better way to develop a runtime change would be to modify files in `${EYRIE_SRCDIR}` directly :)